fog/distribution/src/main.rs

```diff
@@ -369,7 +372,14 @@ fn worker_thread_entry(
         txs_created += 1;
 
         // Submit tx
-        if submit_tx(txs_created, &conns, &tx, &config, &logger) {
+        if submit_tx(
+            txs_created,
+            &conns,
+            &tx,
+            &config,
+            &logger,
+            &spendable_txouts_sender,
+        ) {
```

Comment:
So I want to suggest another direction:

It looks that `submit_tx` currently returns `true` if it succeeded, and `false` if it fails.
(Ideally this would be documented)

One option would be, if it fails, then this function puts the SpendableTxOut's back in the queue.

And then we don't need the `rebuild_tx` function at all, right?